tools/accuracy_checker/accuracy_checker/representation/segmentation_representation.py

```diff
@@ -63,14 +64,17 @@ def to_polygon(self, segmentation_colors=None):
             contours, _ = cv.findContours(binary_mask, cv.RETR_TREE, cv.CHAIN_APPROX_SIMPLE)
 
             for contour in contours:
-                if contour.size >= 6:
-                    contour = np.squeeze(contour, axis=1)
-                    if i not in polygons:
-                        polygons[i] = [contour]
-                    else:
-                        polygons[i].append(contour)
+                if contour.size < 6:
+                    continue
 
-        return polygons
+                contour = np.squeeze(contour, axis=1)
+                polygons.append((i, contour))
+
+        result = defaultdict(list)
```

Comment:
you can directly write to result, no need in polygons variable

Reply:
done